src/database/volatile/mod.rs

```diff
@@ -16,33 +17,44 @@ use crate::{
     model::State,
 };
 
-pub struct Database<'a> {
-    memory: HashMap<State, &'a [u8]>,
+/// The Database struct holds an in-memory hashmap from States to BitVec values (derived from Records)
+pub struct Database {
+    memory: HashMap<State, BitVec<u8, Msb0>>,
 }
 
-impl Database<'_> {
+impl Database {
+    /// The database is initialized with an empty hashmap.
     pub fn initialize() -> Self {
         Self {
             memory: HashMap::new(),
         }
     }
 }
 
-impl KVStore for Database<'_> {
+/// The KVStore implementation includes three functions:
+/// - put: takes in a State "key" and a Record "value", and stores the Record's raw value in its HashMap.
+/// - get: takes in a State "key" and returns the raw value stored under that key as a BitSlice.
+/// - del: takes in a State "key" and removes any value stored under that key.
+impl KVStore for Database {
     fn put<R: Record>(&mut self, key: State, value: &R) {
-        todo!()
+        let new = BitVec::from(value.raw()).clone();
+        self.memory.insert(key, new);
     }
 
     fn get(&self, key: State) -> Option<&BitSlice<u8, Msb0>> {
-        todo!()
+        let vecOpt = self.memory.get(&key);
```

Comment:
It might be helpful to use the idiom `if let Some(vector) = self.memory.get(&key)`